src/handlers/rfqm_handlers.ts

```diff
@@ -111,6 +141,77 @@ export class RfqmHandlers {
         res.status(HttpStatus.OK).send(firmQuote);
     }
 
+    public async submitSignedQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
+        RFQM_SIGNED_QUOTE_SUBMITTED.inc();
+        const params = parseSubmitSignedQuoteParams(req);
+
+        if (params.type === RfqmTypes.MetaTransaction) {
+            const metaTransactionHash = params.metaTransaction.getHash();
+
+            // check that the firm quote is recognized as a previously returned quote
+            const quote = await this._rfqmService.dbUtils.findQuoteByMetaTransactionHashAsync(metaTransactionHash);
+            if (quote === undefined) {
+                RFQM_SIGNED_QUOTE_NOT_FOUND.inc();
+                throw new NotFoundError(`metaTransaction quote not found`);
+            }
+
+            // validate that the firm quote is fillable using the origin registry address (this address is assumed to hold ETH)
+            try {
+                await this._rfqmService._blockchainUtils.validateMetaTransactionOrThrowAsync(
```

Comment:
Does this also check that the expiration window is reasobale?

Reply:
adding that to RfqmService